### PR TITLE
Issue #260: Fix IllegalArgumentException with GlobalVar taking a single argument

### PR DESCRIPTION
**Pull request by Alex-Vol-SV**
When calling a GlovalVar declared to accept one argument and the caller
passes null to the function it yields an IllegalArgumentException
instead of passing the null argument. This is unique to single argument
method calls and appears rooted in the way Grrovy lang handles the
invocation when null is provided.

This fix was inspired from the code in the invoked Groovy methods.

### CODE REVIEW

File: src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy [hunk 1]
```diff
@@ -454,7 +454,16 @@ class PipelineTestHelper {
...
+                        { args ->
+                            // When calling a one argument method with a null argument the
+                            // Groovy doMethodInvoke appears to incorrectly assume a zero
+                            // argument call signature for the method yielding an IllegalArgumentException
+                            if (args == null && m.getNativeParameterTypes().size() == 1) {
+                                m.doMethodInvoke(e.value, MetaClassHelper.ARRAY_WITH_NULL)
+                            } else {
+                                m.doMethodInvoke(e.value, args)
+                            }
+                        })
```
Comment (nre-ableton) [inline src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy:466]:
```suggestion
                        { args -> m.doMethodInvoke(e.value, args ?: MetaClassHelper.ARRAY_WITH_NULL) })
```

The Elvis operator is a bit more succinct here.

File: src/test/resources/libs/commons@feature/vars/oneArg.groovy [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// file: vars/oneArg.groovy
+def call(arg) {
+    if (arg == null) {
+        echo "Is null"
+    } else {
+        echo "Is not null"
+    }
```
Comment (nre-ableton) [inline src/test/resources/libs/commons@feature/vars/oneArg.groovy:7]:
```suggestion
    echo arg ? "Is not null" : "Is null"
```

This can also be condensed into a one-liner.

File: src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy (hunk 1 above)
Comment (Alex-Vol-SV) [reply src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy:466]: I do not believe it is for all use cases. The additional check for number of parameters is needed. args may be null for a zero argument method call.

Comment (nre-ableton) [reply src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy:466]: Ah, I see. Shouldn't it be `m.getNativeParameterTypes().size() > 1` then?

Comment (Alex-Vol-SV) [reply src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy:466]: In my opinion this code in Groovy is incorrectly handling the `null` case
[invoked Groovy](https://github.com/apache/groovy/blob/master/src/main/java/org/codehaus/groovy/reflection/ParameterTypes.java#L161)

The line in question should assign EMPTY_ARRAY to the null argumentArray and allow the subsequent lines decide if ARRAY_WITH_NULL is instead appropriate.

Comment (Alex-Vol-SV) [reply src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy:466]: No, more than one null arguments would have manifested as array of null. I can add additional tests for the behavior with multiple arguments but I believe the null single argument is an edge case not handled properly.

Comment (nre-ableton) [reply src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy:466]: Ok, I'm a bit unfamiliar with the background of the original failure here. I need to sit down and read through the linked issue before approving this PR. However, if another maintainer sees that everything is alright, I'm fine with them merging this (hence my leaving neutral feedback and not "requested changes").
